[MRG] Add drive strength plotting

dikshant182004 opened the pull request:
i have removed the unnecessary dependecies and added an example in plot_connectivity.py via taking some pre defined external drives and also added one for convenience .i have also updated the unit section ,the test case is pending

asoplata commented:
Dikshant, please do not close PRs and then open new ones for the same Issue or feature; instead, make your changes to the pre-existing PRs. When you make further changes for this feature, please make them to this PR only.

dikshant182004 commented:
Actually i was facing problem in previous branch thatswhy i had to create new one for contribution

dikshant182004 commented:
is something i need to do in my code ?

dikshant182004 commented:
i have added the test case of depicting the strength of empty drive in the plot ,and it is working perfectly fine however i was not able to see the figure due to the canvas warning 
UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
But its working fine and i have added the example in plot connectivity.py with added external drive ,this is the output 
![Screenshot (970)](https://github.com/user-attachments/assets/f8839e8a-1e61-483f-906d-2df3fc6d539a)
.do have a look

dikshant182004 commented:
??

asoplata commented:
Hello Dikshant, I have finally had time to review the code. I pushed a new commit to your branch with some relatively minor changes; they are described in the commit message. I've also activated CI testing in preparation for merging and finishing this PR. Thanks a ton for your work!

If you have any issues with the changes in my commit, or want to make any further changes yourself, let me know. Once you tell me that you are okay with the current version, I will merge it. Thanks again for your contributions!

dikshant182004 commented:
why merging is blocked ?
